### PR TITLE
Make return type of IResource.getAdapter(Class) generic and rename it to adaptTo(Class)

### PR DESCRIPTION
**Pull request by srossbach**
This patch is mainly an updated version of patch
https://saros-build.imp.fu-berlin.de/gerrit/#/c/3291/1

without actually changing the method name of getAdapter

### CODE REVIEW

**Comment by srossbach** [comment]
You can take over @tobous 

BTW The IntelliJ project cannot be compiled under Eclipse.

private final SelectionListener selectionListener = this::generateSelectionActivity;

ERROR The target type of this expression must be a functional interface

**Comment by tobous** [comment]
> BTW The IntelliJ project cannot be compiled under Eclipse.

@srossbach I am not familiar with the Eclipse workflow. What do you mean by compiling the IntelliJ project under Eclipse? Can you tell me what exactly you did so that I can try to re-create the failure?

**Comment by srossbach** [comment]
Checkout the projects. Call gradle with prepareEclipse. Just do clean and rebuild. Regarding the error message it seems that the compiler think that the SelectionListener isn't a functional interface. However it only has one method so it should be one.

**Comment by srossbach** [comment]
Seems I also have other issues:

Incompatible .class files version in required binaries. Project 'de.fu_berlin.inf.dpp.intellij' is targeting a 1.8 runtime, but is compiled against 'Z:gradle/caches/modules-2/files-2.1/com.jetbrains.intellij.idea/ideaIC/LATEST-EAP-SNAPSHOT/108e4b12d647d8a0cba9a8e557d84c3024c90dd0/ideaIC-LATEST-EAP-SNAPSHOT/lib/jshell-frontend.jar' which requires a 1.9 runtime

**Comment by tobous** [comment]
Rebased onto current master.

Updated commit message.

Removed commented out code in `EclipseResourceImpl.getAdapter(Class)`.

Removed unnecessary parentheses around getAdapter(Class) call in Saros/I `CollaborationUtils`.

**Comment by tobous** [comment]
Amends `[API][CORE] Make return type of IResource.getAdapter() generic` to remove the casting from usages of `getAdapter(Class)` in comments.

Adds commit renaming `IResource.getAdapter(Class)` to `IResource.adaptTo(Class)`.

File: de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/model/roster/RosterAdapterFactory.java [hunk 1]
```diff
@@ -13,8 +13,8 @@
  * <p>E.g. let's you adapt {@link RosterGroupElement} to {@link RosterGroup} with
  *
  * <pre>
- * RosterGroup rosterGroup = (RosterGroup) rosterGroupElement
- *     .getAdapter(RosterGroup.class);
+ * RosterGroup rosterGroup = rosterGroupElement.adaptTo(RosterGroup.class);
```
Comment (srossbach) [inline de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/model/roster/RosterAdapterFactory.java]: Nope, this the ISelectionService interface

Comment (tobous) [reply de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/model/roster/RosterAdapterFactory.java]: My bad. Forgot that the refactoring logic can't differentiate methods in plaintext comments. Will pay more attention in the future.
Fixed.

**Comment by tobous** [comment]
Rebased onto master.

Renamed [`IResourceTest.testGetAdapter()`](https://github.com/saros-project/saros/blob/master/de.fu_berlin.inf.dpp/test/junit/de/fu_berlin/inf/dpp/filesystem/IResourceTest.java#L11) to `testAdaptTo()` to match new method name.

**Pull request merged**